wallet/src/actors/app/state.rs

```diff
@@ -173,7 +173,11 @@ impl State {
     }
 
     /// Updates the node state
-    pub fn update_node_state(&mut self, node_state: StateMachine) {
-        self.node_state = Some(node_state);
+    pub fn update_node_state(&mut self, node_state: Option<StateMachine>) {
+        if node_state == None {
+            self.node_state = None;
```

Comment:
This can be rewritten as self.node_state = node_state. But we can check whether we already can change it from app/methods, thus avoiding this function